src/redux/actions.js

```diff
@@ -0,0 +1,28 @@
+import axios from 'axios';
+
+export const setCountryName = (name) => ({
+  type: 'SET_COUNTRY_NAME',
+  payload: name,
+});
+
+export const setCountryData = (data) => ({
+  type: 'SET_COUNTRY_DATA',
+  payload: data,
+});
+
+export const fetchCountryData = (countryName) => async (dispatch) => {
+  try {
+    const response = await axios.get(
+      `https://restcountries.com/v2/name/${countryName}`,
+    );
+    const { data } = response;
+
+    if (data.length === 0) {
+      dispatch(setCountryData([]));
+    } else {
+      dispatch(setCountryData(data));
+    }
+  } catch (error) {
+    dispatch(setCountryData([]));
+  }
+};
```

Comment:
**OPTIONAL** - You have done well so far, but I suggest that you use the redux toolkit for your redux store, the createSlice and createAsyncThunk hooks will help you avoid a lot of boilerplate code that comes with using redux. Please fix this 🙏